### PR TITLE
EDU-929 Update node and dependencies

### PR DESCRIPTION
**Pull request by ahelmberger**
Closes https://educandu.atlassian.net/browse/EDU-929

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -19,15 +19,15 @@ jobs:
       - name: Install node
         uses: actions/setup-node@v2
         with:
-          node-version: "16.13.1"
+          node-version: "18.12.1"
```
Comment (adrianaluchian) [inline .github/workflows/publish.yml:22]: So far we've (at least mostly?) been less strict here, locking the major version only (`^18.12.1`)

File: .github/workflows/verify.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ jobs:
       - name: Install node
         uses: actions/setup-node@v2
         with:
-          node-version: "16.13.1"
+          node-version: "18.12.1"
```
Comment (adrianaluchian) [inline .github/workflows/verify.yml:17]: Same here.

File: .github/workflows/publish.yml (hunk 1 above)
Comment (ahelmberger) [reply .github/workflows/publish.yml:22]: This is for the Github action. We have to specify the exact version here (and always have). In the `package.json` we (mostly) use `^`.

File: src/lambda/config.js [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
 
+export const SESSION_COOKIE_NAME = process.env.SESSION_COOKIE_NAME;
+if (!SESSION_COOKIE_NAME) {
+  throw new Error('Environment variable SESSION_COOKIE_NAME is not defined');
+}
+
 // The CDN is supposed to have the same protocol as the website!
 export const CLOUDFRONT_PROTO = new URL(WEBSITE_BASE_URL).protocol;
 
-export const SESSION_COOKIE_NAME = process.env.SESSION_COOKIE_NAME || 'SESSION_ID';
```
Comment (adrianaluchian) [inline src/lambda/config.js:12]: Good idea to not default with this variable, one could easily overlook setting it up.

Comment (ahelmberger) [reply src/lambda/config.js:12]: As I did while testing yesterday :rofl:

File: .github/workflows/publish.yml (hunk 1 above)
Comment (adrianaluchian) [reply .github/workflows/publish.yml:22]: I know what the setup is for of course, I just mixed where we locked them and how. Ok 👍

**Review by adrianaluchian — APPROVED** [review]

File: src/lambda/config.js (hunk 3 above)
Comment (adrianaluchian) [reply src/lambda/config.js:12]: Better now than later in actual setup 😅

**Pull request merged**